pkg/internal/lp2ptransports/lp2ptransports.go

```diff
@@ -0,0 +1,63 @@
+package lp2ptransports
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/ipld/go-ipld-prime/codec/dagcbor"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+)
+
+var clog = logging.Logger("lassie:lp2p:tspt:client")
+
+// TransportsProtocolID is the protocol for querying which retrieval transports
+// the Storage Provider supports (http, libp2p, etc)
+const TransportsProtocolID = protocol.ID("/fil/retrieval/transports/1.0.0")
+
+const streamReadDeadline = 30 * time.Second
```

Comment:
long time; I think we have a timeout for retrievals of 20 seconds when using the CLI so it's a bit inconsistent this is so long

Reply:
yea, fair this is a copy from boost we can make it whatever.